public/components/common/query_utils/index.ts

```diff
@@ -205,19 +217,28 @@ export const getIndexPatternFromRawQuery = (query: string): string => {
   return getPromQLIndex(query) || getPPLIndex(query);
 };
 
+function extractSpanAndResolution(query: string) {
+  const regex = /'(\d+)([smhdwMy])'/;
+  const match = query.match(regex);
+  // eslint-disable-next-line radix
+  return match ? { span: parseInt(match[1]), resolution: match[2] } : null;
```

Comment:
specify radix instead of ignoring lint?

and can these functions have tests?

Reply:
added radix, will add test too